### PR TITLE
Resolve followup

### PR DESCRIPTION
**Pull request by udai-kiran**

### CODE REVIEW

File: paperqa/docs.py [hunk 1]
```diff
@@ -893,6 +894,27 @@ async def aquery(
                 memory_str = str(self.memory_model.load_memory_variables({})["memory"])
                 logging.trace(f"trace_id:{trace_id} conversation_history:{memory_str}")
 
+            if(len(self.memory_model.buffer) != 0):
```
Comment (udai-kiran) [inline paperqa/docs.py:897]: this should be if self.memory_model.buffer

File: paperqa/docs.py [hunk 2]
```diff
@@ -893,6 +894,27 @@ async def aquery(
...
+                # print(f"My follow up question:{answer.question}")
+                followup_chain = make_chain(
+                    self.prompts.followup,
+                    cast(BaseLanguageModel, self.llm),
+                    # memory=self.memory_model,
+                    system_prompt=self.prompts.system,
+                )
+                previous_question = self.memory_model.buffer[-2].content
+                # print(f"Previous Question {previous_question}")
```
Comment (udai-kiran) [inline paperqa/docs.py:906]: remove print.

File: paperqa/docs.py [hunk 3]
```diff
@@ -893,6 +894,27 @@ async def aquery(
...
+                    logging.trace(f"trace_id:{trace_id} context:{answer.context}")
+                    new_followup_question = await followup_chain.arun(
+                    previous_question=previous_question,
+                    question=answer.question,
+                    # callbacks=callbacks,
+                    )
+                except Exception as e:
+                    new_followup_question = str(e)
+                answer.question = new_followup_question
+                # print(f"llm follow up question: {answer.question}")
```
Comment (udai-kiran) [inline paperqa/docs.py:917]: remove the print

File: paperqa/docs.py [hunk 4]
```diff
@@ -893,6 +894,27 @@ async def aquery(
...
+                # print(f"Previous Question {previous_question}")
+                try:
+                    logging.trace(f"trace_id:{trace_id} context:{answer.context}")
+                    new_followup_question = await followup_chain.arun(
+                    previous_question=previous_question,
+                    question=answer.question,
+                    # callbacks=callbacks,
+                    )
+                except Exception as e:
+                    new_followup_question = str(e)
```
Comment (udai-kiran) [inline paperqa/docs.py:915]: make the variable followup_question instead of new_followup_questions.

Comment (vedantacharya1998) [reply paperqa/docs.py:915]: Done.

File: paperqa/docs.py (hunk 1 above)
Comment (vedantacharya1998) [reply paperqa/docs.py:897]: Done

**Pull request merged**